compiler/rustc_codegen_llvm/src/gotoc/mod.rs

```diff
@@ -294,8 +354,17 @@ impl CodegenBackend for GotocCodegenBackend {
             let items = cgu.items_in_deterministic_order(tcx);
             for (item, _) in items {
                 match item {
-                    MonoItem::Fn(instance) => c.declare_function(instance),
-                    MonoItem::Static(def_id) => c.declare_static(def_id, item),
+                    MonoItem::Fn(instance) => {
+                        set_debug_item(format!(
+                            "declare_function: {}",
+                            c.readable_instance_name(instance)
+                        ));
+                        c.declare_function(instance)
+                    }
+                    MonoItem::Static(def_id) => {
+                        set_debug_item(format!("declare_static: {:?}", def_id));
```

Comment:
We should reset the item at the end, to avoid spuriously having a set item if there is none.  We could use a RAII pattern, tho that might be overkill

Reply:
Done, reset after both loops.

Reply:
Not sure what exactly RAII would look like here, feel free to follow up if this change is not sufficient?